include/te_compiler.h

```diff
@@ -97,6 +97,18 @@
 #define TE_DEPRECATED
 #endif
 
+#if __has_attribute(constructor)
+/**
+ * Mark a function as a constructor. It will be executed
+ * before the main function.
+ */
+#define TE_CONSTRUCTOR_PRIORITY(n) __attribute__((constructor(n)))
+#define TE_CONSTRUCTOR __attribute__((constructor))
+#else
+#define TE_CONSTRUCTOR_PRIORITY(n)
+#define TE_CONSTRUCTOR
```

Comment:
It is nice, but I don't understand how it works if we fall into the branch.

Reply:
IMHO, it should not result in TE build failure, if the feature is not used.
If these defines are used, it should break the build since it will not work as expected.
The problem is that selftests use it. I guess we can we check it selftests meson and exclude corresponding functionality/components if the feature is not available.